spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/SqlContext.java

```diff
@@ -32,42 +31,29 @@
  */
 class SqlContext {
 
-	private final RelationalPersistentEntity<?> entity;
-	private final Table table;
-
-	SqlContext(RelationalPersistentEntity<?> entity) {
-
-		this.entity = entity;
-		this.table = Table.create(entity.getQualifiedTableName());
-	}
+	Table getTable(AggregatePath path) {
```

Comment:
There's very little comment indicating this thing is either not ready for prime-time or the concepts behind aren't clear.

Reply:
Historically `SqlContext` was about getting from entity information to `Table` and `Column`. 
A lot of that moved to `TableInfo`, `ColumnInfo`, and `ColumnInfos`.
With recent changes triggered by your comments the later actually creates (and memoizes) `Column` instances.
We probably could move all of `SqlContext` to these classes.

I think, I avoided this so far because the -Info classes operated on `SqlIdentifier` level so far. 

What do you think?